testhelpers/assertions.go

```diff
@@ -122,6 +137,155 @@ func (a AssertionManager) Contains(actual, expected string) {
+	if err != nil {
+		a.testObject.Fatalf(
+			"Unable to unmarshal 'expectedJSON': %q", err,
+		)
+	}
+
+	if !comparehelpers.DeepContains(actual, expected) {
+		expectedJSONDebug, err := json.Marshal(expected)
+		if err != nil {
+			a.testObject.Fatalf("unable to render expected failure expectation: %q", err)
+		}
+
+		actualJSONDebug, err := json.Marshal(actual)
+		if err != nil {
+			a.testObject.Fatalf("unable to render actual failure expectation: %q", err)
+		}
+
+		var prettifiedExpected bytes.Buffer
+		err = json.Indent(&prettifiedExpected, expectedJSONDebug, "", "  ")
+		if err != nil {
+			a.testObject.Fatal("failed to format expected TOML output as JSON")
+		}
+
+		var prettifiedActual bytes.Buffer
+		err = json.Indent(&prettifiedActual, actualJSONDebug, "", "  ")
+		if err != nil {
+			a.testObject.Fatal("failed to format actual TOML output as JSON")
+		}
+
+		actualJSONDiffArray := strings.Split(prettifiedActual.String(), "\n")
+		expectedJSONDiffArray := strings.Split(prettifiedExpected.String(), "\n")
+
+		a.testObject.Fatalf(
+			"Expected '%s' to contain '%s'\n\nJSON Diff:%s",
+			prettifiedActual.String(),
+			prettifiedExpected.String(),
+			cmp.Diff(actualJSONDiffArray, expectedJSONDiffArray),
+		)
+	}
+}
+
+func (a AssertionManager) ContainsYAML(actualYAML, expectedYAML string) {
+	a.testObject.Helper()
+
+	var actual interface{}
+	err := yaml.Unmarshal([]byte(actualYAML), &actual)
+	if err != nil {
+		a.testObject.Fatalf(
+			"Unable to unmarshal 'actualJSON': %q", err,
+		)
+	}
+
+	var expected interface{}
+	err = yaml.Unmarshal([]byte(expectedYAML), &expected)
+	if err != nil {
+		a.testObject.Fatalf(
+			"Unable to unmarshal 'expectedYAML': %q", err,
+		)
+	}
+
+	if !comparehelpers.DeepContains(actual, expected) {
+		expectedYAMLDebug, err := yaml.Marshal(expected)
+		if err != nil {
+			a.testObject.Fatalf("unable to render expected failure expectation: %q", err)
+		}
+
+		actualYAMLDebug, err := yaml.Marshal(actual)
+		if err != nil {
+			a.testObject.Fatalf("unable to render actual failure expectation: %q", err)
+		}
+
+		actualYAMLDiffArray := strings.Split(string(actualYAMLDebug), "\n")
+		expectedYAMLDiffArray := strings.Split(string(expectedYAMLDebug), "\n")
+
+		a.testObject.Fatalf(
+			"Expected '%s' to contain '%s'\n\nDiff:%s",
+			string(actualYAMLDebug),
+			string(expectedYAMLDebug),
+			cmp.Diff(actualYAMLDiffArray, expectedYAMLDiffArray),
+		)
+	}
+}
+
+func (a AssertionManager) ContainsTOML(actualTOML, expectedTOML string) {
+	a.testObject.Helper()
+
+	var actual interface{}
+	err := toml.Unmarshal([]byte(actualTOML), &actual)
+	if err != nil {
+		a.testObject.Fatalf(
+			"Unable to unmarshal 'actualTOML': %q", err,
+		)
+	}
+
+	var expected interface{}
+	err = toml.Unmarshal([]byte(expectedTOML), &expected)
+	if err != nil {
+		a.testObject.Fatalf(
+			"Unable to unmarshal 'expectedTOML': %q", err,
+		)
+	}
+
+	// TODO: convert to a toml based diff. Unfortunately marshalling toml
```

Comment:
Complete TODO.

Reply:
Think this one might be a bit out of scope, for now. But should be addressed when there is a `toml` library that supports uninstalling into an arbitrary `interface`. Should I just remove the `TODO` part of this?

Reply:
The TODO comment didn't make too much sense to me to begin with. Based on your comment, I would agree that it's probably not something worth having a TODO. I'm all for deleting it.